src/core/mac/mac_frame.cpp

```diff
@@ -777,48 +845,18 @@ uint8_t Frame::SkipSecurityHeaderIndex(void) const
         break;
 
     case kFcfSrcAddrShort:
-        index += sizeof(ShortAddress);
+        offset += sizeof(ShortAddress);
         break;
 
     case kFcfSrcAddrExt:
-        index += sizeof(ExtAddress);
+        offset += sizeof(ExtAddress);
         break;
 
     default:
-        ExitNow(index = kInvalidIndex);
-    }
-
-    VerifyOrExit((index + GetFcsSize()) <= GetPsduLength(), index = kInvalidIndex);
-
-    // Security Control + Frame Counter + Key Identifier
-    if ((fcf & kFcfSecurityEnabled) != 0)
-    {
-        uint8_t securityControl = *(GetPsdu() + index);
-
-        index += kSecurityControlSize + kFrameCounterSize;
-
-        switch (securityControl & kKeyIdModeMask)
-        {
-        case kKeyIdMode0:
-            index += kKeySourceSizeMode0;
-            break;
-
-        case kKeyIdMode1:
-            index += kKeySourceSizeMode1 + kKeyIndexSize;
-            break;
-
-        case kKeyIdMode2:
-            index += kKeySourceSizeMode2 + kKeyIndexSize;
-            break;
-
-        case kKeyIdMode3:
-            index += kKeySourceSizeMode3 + kKeyIndexSize;
-            break;
-        }
+        break;
```

Comment:
Keep the fault handling code.
```suggestion
        ExitNow(index = kInvalidIndex);
```